coordinator/src/manyprover.rs

```diff
@@ -0,0 +1,275 @@
+    let mut benchmark_out = match &input.benchmark_output {
+        Some(benchmark_config) => {
+            match BenchmarkingOutput::from_config(
+                benchmark_config.clone(),
+                input.estimate_expected_number_proofs(),
+            )
+            .await
+            {
+                Ok(benchmark_output) => Some(benchmark_output),
+                Err(err) => {
+                    error!("Failed to construct Benchmark Output: {}", err);
+                    return Err(ManyProverError::BenchmarkingOutputBuild(err));
+                }
+            }
+        }
+        None => {
+            info!("Was not provided means to place benchmarking statistics output...");
+            None
+        }
+    };
+
+    // Stores the previous PlonkyProofIntern if applicable
+    let mut prev: Option<PlonkyProofIntern> = None;
+    // Stores the current block number
+    let mut cur_block_num: u64 = input.start_block_number;
+
+    //========================================================================
+    // Performing the proofs
+    //========================================================================
+
+    info!("Starting the proof process");
+    let benchmark_start_instance = Instant::now();
+
+    loop {
+        // Determine if we should begin a proof depending on the termination policy.
+        match input.terminate_on {
+            Some(TerminateOn::ElapsedSeconds {
+                num_seconds,
+                include_straddling: _,
+            }) => {
+                if benchmark_start_instance.elapsed().as_secs() >= num_seconds {
+                    info!(
+                        "Terminating on block {} due to reaching time constraint.",
+                        cur_block_num
+                    );
+                    break;
+                }
+            }
+            Some(TerminateOn::EndBlock { block_number }) if cur_block_num > block_number => {
+                info!(
+                    "Terminating on block {} as we have surpassed the end block.",
+                    cur_block_num
+                );
+                break;
+            }
+            Some(TerminateOn::NumBlocks { num_blocks })
+                if cur_block_num >= (input.start_block_number + num_blocks) =>
+            {
+                info!(
+                    "Terminating on block {} as we have proved {} number of blocks",
+                    cur_block_num, num_blocks
+                );
+                break;
+            }
+            _ => {
+                info!("Starting attempt to prove block {}", cur_block_num);
+            }
+        }
+
+        //------------------------------------------------------------------------
+        // Fetching
+        //------------------------------------------------------------------------
+
+        debug!("Attempting to fetch block {}", cur_block_num);
+        let fetch_start_instance = Instant::now();
+        let prover_input = match fetch(
```

Comment:
This `match` is hard to read because the function name is so small and the argument list is so long. At first I thought you were matching on a 3-tuple of:
```
(
  cur_block_num,
  input.checkpoint_block_number,
  &input.block_source
)
  ```
  but then I realized that you were matching on the return value of `fetch()`.
  
  Anyway - I think it makes code maintenance easier if you separate function calls and `match` statements, like this:
  ```
  let fetched = fetch(...);
  let prover_input = match fetched {
      ...
  };
  ```
  
But feel free to ignore this feedback for this specific case, since we may not be maintaining this code in the long-term. Just something to keep in mind for the future.

Also, there are a number of these throughout the code, but I'll just comment on this one.